### PR TITLE
rawPayload shouldn't return the full reply buffer

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

**Comment by dothebart** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/6888/

File: lib/Rest/VstRequest.cpp [hunk 1]
```diff
@@ -66,6 +66,11 @@ VstRequest::VstRequest(ConnectionInfo const& connectionInfo,
   parseHeaderInformation();
 }
 
+arangodb::velocypack::StringRef VstRequest::rawPayload() const {
+  return arangodb::velocypack::StringRef(reinterpret_cast<const char*>(_buffer.data() + _payloadOffset),
```
Comment (graetzer) [inline lib/Rest/VstRequest.cpp]: add a check that the buffer is bigger than the offset

File: lib/Rest/VstRequest.cpp [hunk 2]
```diff
@@ -66,6 +66,15 @@ VstRequest::VstRequest(ConnectionInfo const& connectionInfo,
   parseHeaderInformation();
 }
 
+arangodb::velocypack::StringRef VstRequest::rawPayload() const {
+  if (_buffer.size() < _payloadOffset) {
```
Comment (jsteemann) [inline lib/Rest/VstRequest.cpp]: Changing this to
```suggestion
  if (_buffer.size() <= _payloadOffset) {
```
will be even better, though the previous approach will already prevent disaster.

**Comment by dothebart** [comment]
tests green.

**Pull request merged**